ignite/trainer/trainer.py

```diff
@@ -212,3 +213,41 @@ def run(self, training_data, max_epochs=1):
             self._logger.error("Training is terminating due to exception: %s", str(e))
             self._fire_event(TrainingEvents.EXCEPTION_RAISED)
             raise e
+
+
+def create_supervised(model, optimizer, loss_fn, cuda=False):
+    """
+    Factory function for creating a trainer for supervised models
+
+    Args:
+        model (torch.nn.Module): the model to train
+        optimizer (torch.optim.Optimizer): the optimizer to use
+        loss_fn (torch.nn loss function): the loss function to use
+        cuda (bool, optional): whether or not to transfer batch to GPU (default: False)
+
+    Returns:
+        Trainer: a trainer instance with supervised update and inference functions
+    """
+    def _prepare_batch(batch):
```

Comment:
Perhaps it would be nice to have a `training` flag here, to use `Variable(..., volatile=True)` (`torch` `0.3`)  or `torch.autograd.no_grad()`  (`torch` `0.4`) during inference?